event-store/src/main/java/org/hypertrace/core/eventstore/EventConsumer.java

```diff
@@ -1,41 +1,32 @@
 package org.hypertrace.core.eventstore;
 
 import java.time.Duration;
+import java.util.List;
 
 /**
  * While this is similar to Kafka consumer. The key difference is that the consumer is initialized
  * per topic unlike Kafka where one can consume events across all topics.
  */
-public interface EventConsumer<T> {
+public interface EventConsumer<K, V> {
```

Comment:
Do we care about backwards compatibility where the old interface would be used by some of the dependents and they would spit out data to be consumed by ones using the new API?

Reply:
From kafka perspective, key and value type are always needed. In the previous version, <K> was hard-coded to byte[] while <V> was being provided.
If someone moves to new version of event store, then they will have explicitly pass byte[] for <K>. Apart from that there will be no compatibility either on consumer or producer side.

Reply:
>Apart from that there will be no compatibility either on consumer or producer side.

You mean there's no issue on a consumer using the old version and a producer using the new one?

Reply:
Yes, they will work without any issue.

Reply:
ok.

Reply:
:( we always care about backwards compatibility about APIs! not just between producers and consumers, but between multiple modules in the same JVMs.